html/browsers/browsing-the-web/navigating-across-documents/008.html

```diff
@@ -1,17 +1,37 @@
 <!doctype html>
-<title>Link with onclick form submit to javascript url and href navigation </title>
+<link rel="help" href="https://html.spec.whatwg.org/multipage/browsing-the-web.html#navigating-across-documents">
+<link rel="help" href="https://html.spec.whatwg.org/multipage/browsing-the-web.html#javascript-protocol">
+<link rel="help" href="https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#concept-form-submit">
+<link rel="help" href="https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#submit-get-action">
+<link rel="help" href="https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#plan-to-navigate">
+<link rel="help" href="https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#planned-navigation">
+<title>Link with onclick form submit to javascript: url is aborted by href navigation</title>
 <script src="/resources/testharness.js"></script>
 <script src="/resources/testharnessreport.js"></script>
 <div id="log"></div>
 <iframe id="test" name="test"></iframe>
-<form target="test" action="javascript:'<script>parent.postMessage(&quot;click&quot;, &quot;*&quot;)</script>'"></form>
+<form target="test" action="javascript:parent.postMessage('click', '*')"></form>
 <a target="test" onclick="document.forms[0].submit()" href="href.html">Test</a>
 <script>
-var t = async_test();
-t.step(function() {document.getElementsByTagName("a")[0].click()});
-onmessage = t.step_func(
-  function(e) {
-    assert_equals(e.data, "href");
-    t.done();
-  });
+var test = async_test();
+test.step(function() {
+  document.querySelector("a").click();
+});
+window.onmessage = test.step_func(function(e) {
+  assert_equals(e.data, "href");
+  test.done();
+});
 </script>
+
+<!--
+1. This test creates an iframe, a form with
+    `action="javascript:parent.postMessage('click', '*')", and an anchor
+    with both an `onclick` and `href`.
+2. The test "clicks" the anchor, triggering the `onclick` handler operation,
+    which will attempt to submit the form. The form's `action` is a
+    `javascript:` url, which has special handling in step 12
+3. Navigating the iframe to the anchor's `href` will abort the `action`.
```

Comment:
This is the part that is not obvious to me in terms of how it works.  Might need spec clarifications.

What do browsers end up doing here?

Reply:
https://html.spec.whatwg.org/multipage/browsing-the-web.html#navigating-across-documents, Step 6: 

> Cancel any preexisting but not yet mature attempt to navigate browsingContext, including canceling any instances of the fetch algorithm started by those attempts. If one of those attempts has already created and initialized a new Document object, abort that Document also. (Navigation attempts that have matured already have session history entries, and are therefore handled during the update the session history with the new page algorithm, later.)

The form action navigation is started and reaches step 12 of [navigating-across-documents](https://html.spec.whatwg.org/multipage/browsing-the-web.html#navigating-across-documents), which enters **If resource is a request whose url's scheme is "javascript"**, and does "Queue a task to run these "javascript: URL" steps, associated with the active document of browsingContext:". At this point `onclick` returns, and the `href` navigation begins. When it reaches step 6 of [navigating-across-documents](https://html.spec.whatwg.org/multipage/browsing-the-web.html#navigating-across-documents), it should cancel the form action navigation—however I believe there is a spec bug and I plan to write a fix proposal: there is no explicit spec language to remove that previously queued task (for the "javascript: URL").